.github/workflows/ci.yml

```diff
@@ -9,49 +9,56 @@ on:
       - "master"
 
 jobs:
-  build:
-    name: ghc ${{ matrix.ghc }}
+  generateMatrix:
+    name: "Generate matrix from cabal"
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        ghc: ["8.10", "9.0", "9.2"]
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout base repo
+        uses: actions/checkout@v2
+      - name: Extract the tested GHC versions
+        id: set-matrix
+        run: |
+          wget https://github.com/Kleidukos/get-tested/releases/download/v0.1.5.0/get-tested-0.1.5.0-linux-amd64 -O get-tested
+          chmod +x get-tested
+          ./get-tested --ubuntu tree-sitter/tree-sitter.cabal >> $GITHUB_OUTPUT
```

Comment:
Hey this is cool

Reply:
Actually in the meantime it gained a GitHub Action. I'll send a follow-up PR. :)